### ISSUE TITLE
It is difficult to compare original and reduced shaders due to differences in braces

### ISSUE THREAD

**Issue by afd** [google/graphicsfuzz#782]
To avoid problems with dangling "else"s, GraphicsFuzz adds braces to all "if"s and "else"s if they do not already have them.

Some of the sample shaders do not use braces in the first place.

This creates a problem where, after a bug has been found and the offending shader reduced, it is hard to compare the reference and reduced-variant shaders, because the latter has additional braces while the former does not.

This could be solved in at least the following ways:

(1) Have gfauto run AddBraces and PrettyPrint to dump the reference shader in a form where it will be formatted as per the variant shader

(2) Adapt all the samples so that they have braces.

While (2) might not be a bad thing to do in its own right, I think (1) is the right way to solve this, because it will also lead to any macros in the reference shader being expanded (and we don't necessarily want to have them pre-expanded in all references).

@paulthomson what do you think?

**Comment by paulthomson** [comment]
Yeah either way sounds good. I think in general we will have to preprocess references for the comparison to make sense. And we will want version-specific donors and references. We may want to switch to having a set of base shaders and then a script that will out put pre-processed references and donors. In this case, we would essentially just be doing (2).

**Comment by paulthomson** [comment]
Oh I just re-read what you said. Yes (2) does sound desirable.

**Comment by afd** [comment]
Opened #787 to add braces during reference preparation, which should solve the problem.
